### PR TITLE
fixed issue with passed data vs external data

### LINKED ISSUE #263: Using "var" as the second argument inside "some" will result incorrect outcome
**Environment**
- Nuget Package: JsonLogic.Net, 1.1.11
- Nuget Version: 6.1.0
- OS: WSL
- .Net Target: net6.0

**Describe the bug**
Looks like if you using "var" for a second argument inside of "some" it's returning wrong result. For instance from the attached screenshots from the nice Jsonlogic playground:
image
image

**To Reproduce**
(please refer to the screenshots, I've been using the latest library and the playground should be doing the same)

**Expected behavior**
The first "in" operator case should return "True", as 1 shall be found in [1] during the iteration; then the "greater than" case shall return "False".

**Additional context**
Is it anything like I am using it wrong?

### PR DESCRIPTION
**Pull request by gregsdennis**
Resolves #263 

@wangh-trimble I think this addresses the issue.  It basically follows your suggestion of adding the secondary parameter, but I did it without an overload.  Either way would have been a breaking change, I think.  It's fine.

It passes all of the tests, including your new one, so I guess it's fine.

This functionality overall has been... weird... to implement.

### CODE REVIEW

**Comment by wangh-trimble** [comment]
Thanks for the quick action, really appreciated it! 

Maybe also to add this test case in https://github.com/gregsdennis/json-everything/blob/master/JsonLogic.Tests/SomeTests.cs?

And maybe also for AllRule, NoneRule.
```
[Test]
    public void SomeInTest()
    {
        var rule = new SomeRule(new VariableRule("x"),
            new InRule(new VariableRule(""), new VariableRule("y")));

        var data = JsonDocument.Parse("{ \"x\": [-1, 0, 1], \"y\": [2, 3, 1] }").RootElement;
        JsonAssert.IsTrue(rule.Apply(data));
    }
```

Also can I ask, when this change can be available on Nuget?

File: JsonLogic/Rules/SomeRule.cs [hunk 1]
```diff
@@ -16,15 +16,15 @@ public SomeRule(Rule input, Rule rule)
...
 	{
-		var input = _input.Apply(data);
+		var input = _input.Apply(data, contextData);
 
 		if (input.ValueKind != JsonValueKind.Array)
 			throw new JsonLogicException("Input must evaluate to an array.");
 
 		var inputData = input.EnumerateArray();
-		return inputData.Select(value => _rule.Apply(value))
+		return inputData.Select(value => _rule.Apply(data, value))
```
[the remark this answers is not in the document]
Comment (gregsdennis) [reply JsonLogic/Rules/SomeRule.cs:27]: good catch

**Comment by gregsdennis** [comment]
I did add the test case, except it's in GitHubTests because it relates to a reported issue.

It'll be available a few minutes after the PR is merged.

**Comment by wangh-trimble** [comment]
@gregsdennis Thank you very much for the swift action! Sorry been busy so I haven't fully read your comments (and questions) on the issue side, will check tomorrow when have some more time.

**Pull request merged**